File: .claude/skills/sast-scan/rules/semgrep/ruby/tests/test_cwe78_cmdi.rb
# CWE-78: Command Injection
def test_command_injection(user_input)
  # ruleid: ruby.security.command-injection
  system("ls #{user_input}")
  # ruleid: ruby.security.command-injection
  exec("cat #{user_input}")
  # ruleid: ruby.security.command-injection
  `ping #{user_input}`

  # ok: ruby.security.command-injection
  system("ls", "-la")
  # ok: ruby.security.command-injection
  system("echo", "hello")
end

File: .claude/skills/sast-scan/rules/semgrep/php/tests/test_cwe78_cmdi.php
<?php
// PHP SAST test fixtures — per-CWE coverage
// CWE-78: Command Injection

function testCommandInjection(string $userInput) {
    // ruleid: php.security.command-injection-exec
    exec($userInput);
    // ruleid: php.security.command-injection-exec
    system("ping " . $userInput);
    // ruleid: php.security.command-injection-exec
    shell_exec("nslookup " . $userInput);
    // ruleid: php.security.command-injection-exec
    passthru($userInput);

    // ok: php.security.command-injection-exec
    exec(escapeshellcmd($userInput));
    // ok: php.security.command-injection-exec
    system("ls -la");
    // ok: php.security.command-injection-exec
    exec("echo 'safe'");
}
